### ISSUE TITLE
Can I assign IPAMBlocks for special nodes manually?

### ISSUE THREAD

**Issue by bluven** [projectcalico/calico#8315]
I want to assign IPAMBlocks for some nodes manually in my cluster which is a KubeEdge cluster. A KubeEdge cluster normally have a lot of edge nodes on which calico cannot work. 

After reading calico's source code, it seems this method [EnsureBlock](https://github.com/projectcalico/calico/blob/master/libcalico-go/lib/ipam/interface.go#L107C2-L107C13) can do the job.

I just have some questions:

1. Can I assign IPAMBlocks manually? I don't want to make calico disfunction.
2.  What is the proper way to reclaim a block( I used kubectl delete, but it seems this is not right way)?
3. To call EnsureBlock, an  `ipam.HostReservedAttr` argument is needed.
```go 
	rsvdAttrWindows := &ipam.HostReservedAttr{
		StartOfBlock: 0,
		EndOfBlock:   31,
		Handle:       "edge",
		Note:         "edge host rsvd",
	}
```
I want to know what are the valid values for Handle, StartOfBlock and EndOfBlock? Should handle be unique for each assign?
4. Since calico cannot work on those nodes, I have to assign IP myself. Do I have to manage Unallocated and Allocations fields of IPAMBlocks of those nodes?

Thanks in advance.

**Comment by caseydavenport** [comment]
The IPAMBlock API isn't really intended to be used directly, but obviously with some care you should be able to do it.

Have you looked at simply assigning IP pools to those nodes? The IPPool API includes a node selector that would allow you to allocate certain ranges of IPs to certain nodes - this is the intended way to satisfy your use-case.

>  Do I have to manage Unallocated and Allocations fields of IPAMBlocks of those nodes?

How do you plan to allocate IPs from these blocks? Would these nodes be using the Calico CNI plugin?

**Comment by bluven** [comment]
> Would these nodes be using the Calico CNI plugin?

@caseydavenport No, calico-node won't run on edge nodes, [cni plugins](https://github.com/containernetworking/plugins) will do the assign IP to pods.  I just need calico to assign IPAMBlock for edge nodes. 

> Have you looked at simply assigning IP pools to those nodes?
As I know to assign IPs to pods, an calico-node must be involved, but in my scenario, edge nodes won't have calico-nodes running.

**Comment by caseydavenport** [comment]
I'm curious why you need IPAM blocks on those nodes if Calico won't be running on them at all?

**Comment by bluven** [comment]
> I'm curious why you need IPAM blocks on those nodes if Calico won't be running on them at all?

In fact I implemented a semi-CNI plugin which runs on those edge nodes, previously I choosed to assign PodCIDR for each node , but it requires another CIDR to keep calico safe. It worked, but I just wonder if can use calico client to ask calico to assign IPAMBlock for those nodes.

But as you replied in another issue, the IPAMBlock is internal resource, maybe I should leave it alone.

Thanks for your help.
